src/common/k8s-api/endpoints/cron-job.api.ts

```diff
@@ -44,61 +51,33 @@ export class CronJobApi extends KubeApi<CronJob> {
   }
 }
 
-export interface CronJob {
-  spec: {
-    schedule: string;
-    concurrencyPolicy: string;
-    suspend: boolean;
-    jobTemplate: {
-      metadata: {
-        creationTimestamp?: string;
-        labels?: {
-          [key: string]: string;
-        };
-        annotations?: {
-          [key: string]: string;
-        };
-      };
-      spec: {
-        template: {
-          metadata: {
-            creationTimestamp?: string;
-          };
-          spec: {
-            containers: IPodContainer[];
-            restartPolicy: string;
-            terminationGracePeriodSeconds: number;
-            dnsPolicy: string;
-            hostPID: boolean;
-            schedulerName: string;
-          };
-        };
-      };
-    };
-    successfulJobsHistoryLimit: number;
-    failedJobsHistoryLimit: number;
-  };
-  status: {
-    lastScheduleTime?: string;
-  };
+export interface CronJobSpec {
+  concurrencyPolicy?: string;
+  failedJobsHistoryLimit?: number;
+  jobTemplate?: JobTemplateSpec;
+  schedule: string;
+  startingDeadlineSeconds?: number;
+  successfulJobsHistoryLimit?: number;
+  suspend?: boolean;
 }
 
-export class CronJob extends KubeObject {
-  static kind = "CronJob";
-  static namespaced = true;
-  static apiBase = "/apis/batch/v1beta1/cronjobs";
+export interface CronJobStatus {
+  lastScheduleTime?: string;
+  lastSuccessfulTime?: string;
+  active?: ObjectReference[];
+}
 
-  constructor(data: KubeJsonApiData) {
-    super(data);
-    autoBind(this);
-  }
+export class CronJob extends KubeObject<CronJobStatus, CronJobSpec, "namespace-scoped"> {
+  static readonly kind = "CronJob";
+  static readonly namespaced = true;
+  static readonly apiBase = "/apis/batch/v1beta1/cronjobs";
 
   getSuspendFlag() {
-    return this.spec.suspend.toString();
+    return (this.spec.suspend ?? false).toString();
```

Comment:
Do we really want to return `"false"`? Comparison of booleans turned into strings is not welcome.

Reply:
We do actually. This function (for what it is worth) is really a `renderSuspendFlag()` function. A legacy part of the codebase from before Mirantis IIRC. And the value `false` renders nothing within React, whereas the value `"false"` does render something.

And according to the kube documentation the default value for this field (if not specified) is `false`

Reply:
If you want I can add some documentation for this method so that it is clearer? Or maybe even just depecrate it and move it to a free function?

Reply:
Just wanted to make sure this isn't accident. If you think it's fine, then it's fine.